napalm_nxos/nxos.py

```diff
@@ -61,7 +61,8 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
         if optional_args is None:
             optional_args = {}
 
-        self.transport = optional_args.get('nxos_protocol', 'http')
+        # nxos_protocol is there for backwards compatibility, transport is the preferred method
+        self.transport = optional_args.get('transport', optional_args.get('nxos_protocol', 'http'))
```

Comment:
Let's default to `https` please, as documented - solves https://github.com/napalm-automation/napalm-nxos/issues/36
Then we'll include this PR in a major release, 0.6.0.
Thanks!